### PR TITLE
doc: release: Add ADI and sensor notes for v4.0.0

### PR DESCRIPTION
**Pull request by MaureenHelm**
Adds release notes for:

- ADI SoCs, drivers, and boards
- Sensor drivers

### CODE REVIEW

File: doc/releases/release-notes-4.0.rst [hunk 1]
```diff
@@ -166,6 +166,7 @@ Boards & SoC Support
     versions (1.0 and 1.1) are no longer supported.
   * Added ESP32 WROVER-E-N16R4 variant.
   * STM32H5: Added support for OpenOCD through STMicroelectronics OpenOCD fork.
+  * MAX32: Enabled Segger RTT and SystemView support.
 
 * Added support for these boards:
```
Comment (ttmut) [inline doc/releases/release-notes-4.0.rst:171]: Should max32690fthr be mentioned?

https://github.com/zephyrproject-rtos/zephyr/pull/77285

Comment (kartben) [reply doc/releases/release-notes-4.0.rst:171]: is it not? https://github.com/zephyrproject-rtos/zephyr/blob/65511eacf9d17bf22a6cf28c3ace467a86a159f5/doc/releases/release-notes-4.0.rst?plain=1#L177

**all** boards added for this release have in fact already been contributed to the release notes in a previous commit (https://github.com/zephyrproject-rtos/zephyr/commit/12821f8ec6bb9d817a05fae1e5a51258c1d00d8e)

Comment (kartben) [reply doc/releases/release-notes-4.0.rst:171]: is it not? https://github.com/zephyrproject-rtos/zephyr/blob/65511eacf9d17bf22a6cf28c3ace467a86a159f5/doc/releases/release-notes-4.0.rst?plain=1#L177

**all** boards added for this release have in fact already been contributed to the release notes in a previous commit (https://github.com/zephyrproject-rtos/zephyr/commit/12821f8ec6bb9d817a05fae1e5a51258c1d00d8e)

Comment (ttmut) [reply doc/releases/release-notes-4.0.rst:171]: Oh, I forgot fetch the latest changes. Sorry about that.

**Review by teburd — APPROVED** [review]

**Review by ubieda — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Review by ttmut — APPROVED** [review]

**Pull request merged**